app/controllers/kudos_controller.rb

```diff
@@ -15,10 +15,13 @@ def create
     @kudo = Kudo.new(kudo_params)
     @kudo.giver_id = current_employee.id
     if @kudo.save
+      @current_employee.number_of_available_kudos -= 1
+      @current_employee.save
       flash[:notice] = 'Kudo was created successfully'
       redirect_to root_path
```

Comment:
hej co się stanie jeżeli pomiędzy `@kudo.save` a `@current_employee.save` padnie baza danych?

poczytaj sobie o transakcjach w railsach
np. https://www.paweldabrowski.com/articles/introduction-to-rails-transactions